ext/sockets/sockets.c

```diff
@@ -2274,17 +2293,19 @@ PHP_FUNCTION(socket_addrinfo_lookup)
 
 	for (rp = result; rp != NULL; rp = rp->ai_next) {
 		if (rp->ai_family != AF_UNSPEC) {
-			res = emalloc(sizeof(struct addrinfo));
-			memcpy(res, rp, sizeof(struct addrinfo));
+			zval zaddr;
```

Comment:
Is there any obvious issue with this piece of code? The following few lines result in a segfault, and I can't find out the reason.

Reply:
Looks like you dropped the allocation code for the main addrinfo structure.

Reply:
Ohh, I see it now 🤦  :)